src/main/scala/mesosphere/marathon/core/health/HealthResult.scala

```diff
@@ -7,15 +7,18 @@ sealed trait HealthResult {
   def taskId: Task.Id
   def version: Timestamp
   def time: Timestamp
+  def publishEvent: Boolean
 }
 
 case class Healthy(
   taskId: Task.Id,
   version: Timestamp,
-  time: Timestamp = Timestamp.now()) extends HealthResult
+  time: Timestamp = Timestamp.now(),
+  publishEvent: Boolean = true) extends HealthResult
```

Comment:
Not a fan of adding this to the healthy/unhealthy state. Given the current implementation, this is the smallest invasive change. We need to refactor the health check manager to improve.